### PR TITLE
Create ThriftPacakgeImporter interface

### PR DESCRIPTION
**Pull request by DAddYE**
This diff will make truly public `gen.Generator`.

Currently one of the arguments of `gen.NewGenerator` function contains a private implementation `thriftPackageImporter`. 

This pull request exposes a public `ThriftPackageImporter` instead.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,215 +1,281 @@
...
+
+* Gen: Now the constructor `NewGenerator` use a public interface instead of the
+  private implementation allowing it to be used from others packages.
 
 ## [1.11.0] - 2018-03-27
+
 ### Added
-- Plugins now have access to service and function annotations.
+
+* Plugins now have access to service and function annotations.
```
Comment (abhinav) [inline CHANGELOG.md]: Revert these changes please. We shouldn't be rewriting the changelog as part of this PR.

Also, for your change, all you need to do is add these entries under `Unreleased`. The version number, release date, etc. will be set when this change is actually released.

File: gen/generate.go [hunk 2]
```diff
@@ -154,15 +154,26 @@ func Generate(m *compile.Module, o *Options) error {
...
+	// given Thrift file relative to the ImportPrefix.
+	RelativePackage(file string) (importPath string, err error)
+
+	// RelativePackage returns the relative path of the given Thrift file
+	// relative to the ThriftRoot.
+	RelativeThriftFilePath(file string) (relPath string, err error)
+
+	// Package returns the import path for the top-level package of the given Thrift
+	// file.
+	Package(file string) (ipmortPath string, err error)
```
Comment (abhinav) [inline gen/generate.go]: s/ipmortPath/importPath

File: CHANGELOG.md (hunk 1 above)
Comment (DAddYE) [reply CHANGELOG.md]: done

File: gen/generate.go [hunk 3]
```diff
@@ -154,15 +154,26 @@ func Generate(m *compile.Module, o *Options) error {
 	return nil
 }
 
-// TODO(abg): Make some sort of public interface out of the Importer
+// ThriftPackageImporter determines the import paths from a thrift root
```
[the remark this answers is not in the document]
Comment (DAddYE) [reply gen/generate.go]: Done.

File: gen/generate.go (hunk 2 above)
Comment (DAddYE) [reply gen/generate.go]: Done.

**Comment by DAddYE** [comment]
Hi guys, I've updated the pull request. Can you please take another look? Thanks!
